### PR TITLE
NETWORK_PARAMS - refactor tests and add test for maxBound epoch boundary value

### PR DESCRIPTION
**Pull request by piotr-iohk**
# Issue Number

#1302

# Overview
- 010621576e0cd961313e24a4c669d9dc1631092e
  NETWORK_PARAMS - refactor tests and add test for maxBound epoch bonduary value
  
- 4b6288ded9d1a48e63f848a1463365f521b5baa2
  extract verifyEpochNumOK from tests

# Comments

 - refactored/tagged (`NETWORK_PARAMS`) existing tests to comply with https://docs.google.com/spreadsheets/d/1fadIA_j4nCd3FbylPo5J8K9Fy6tixSC2T9V3xHKuUvU/edit#gid=0
- added test for maxBound epoch value to test boundary value

### CODE REVIEW

**Comment by paweljakubas** [comment]
Why not to extract this part of code in `where` and reuse : 
```
           let endpoint = ( "GET", "v2/network/parameters/"<>arg )
            r <- request @ApiNetworkParameters ctx endpoint Default Empty
            expectResponseCode @IO HTTP.status200 r

            let networkParams = getFromResponse id r
            networkParams `shouldBe` expectedBlockchainParams
```
?

**Comment by piotr-iohk** [comment]
> Why not to extract this part of code...

Done in 4b6288ded9d1a48e63f848a1463365f521b5baa2

File: lib/core-integration/src/Test/Integration/Scenario/API/Network.hs [hunk 1]
```diff
@@ -162,77 +162,92 @@ spec = do
...
+                Link.getNetworkInfo Default Empty
+            let (ApiT (EpochNo currentEpochNo)) =
+                    getFromResponse (#nextEpoch . #epochNumber) r1
+
+            let futureEpochNo = T.pack $ show $ currentEpochNo + 10
+            let endpoint = ( "GET", "v2/network/parameters/"<>futureEpochNo )
+            r2 <- request @ApiNetworkParameters ctx endpoint Default Empty
+
+            expectResponseCode @IO HTTP.status404 r2
+            expectErrorMessage (errMsg404NoEpochNo (T.unpack futureEpochNo)) r2
```
Comment (paweljakubas) [inline lib/core-integration/src/Test/Integration/Scenario/API/Network.hs]: this one also can be reused with `Epoch max value` and others below

**Review by paweljakubas — APPROVED** [review]
LGTM! maybe it would be worth to introduce also `verifyEpochNumWrong ctx epochNo errStatus errMsg`

File: lib/core-integration/src/Test/Integration/Scenario/API/Network.hs (hunk 1 above)
Comment (piotr-iohk) [reply lib/core-integration/src/Test/Integration/Scenario/API/Network.hs]: extracted `verifyEpochNumWrong` in 96bf986e189c3c361937ce5eedc91a01d44daff6.
I think nothing else left/makes really sense to extract :)

**Pull request merged**